lib/spark_api/request.rb

```diff
@@ -107,7 +107,7 @@ def request(method, path, body, options)
     end
     
     def process_request_body(body)
-      if body.is_a?(Hash)
+      if body.is_a?(Hash) || body.is_a?(Array)
```

Comment:
@joshcom: While I believe we have a few other services that support submitting arrays of values, I don't recall this workaround being needed for those. Perhaps the client doesn't have those pulled in yet?

I think the subsequent `{"D":[]}` that would be generated in processing here is what's tripping me up.

Reply:
I'm not sure how those other services are being used, but I can confirm at least the error:
```
[9] pry(main)> SparkApi.put("/newsfeeds/groups", [{"Id":"20140303204140757637000000", SendAgentCopy: false}])                                                            
NoMethodError: undefined method `bytesize' for [{:Id=>"20140303204140757637000000", :SendAgentCopy=>false}]:Array
```

I'm not sure if this is what the front end developers are doing to get around it, but this can be worked around to `.to_json`ing the parameter:

```
[7] pry(main)> SparkApi.put("/newsfeeds/groups", {"D":[{"Id":"20140303204140757637000000", SendAgentCopy: false}]}.to_json) 
=> []
```

Reply:
Hrmmm, so the `{"D":}` wrapper supports an array or a hash on the API-side? I think that was my main blank spot. If that's what the service intends to accept, this PR is fine by me.

Your workaround essentially skips over this chunk of code, so I'd say Annika's fix here is spot on.